python/lsst/meas/algorithms/sourceSelector.py

```diff
@@ -219,8 +229,15 @@ def apply(self, catalog):
             Boolean array indicating for each source whether it is selected
             (True means selected).
         """
-        primary = (catalog[self.primary]*u.nJy).to_value(u.ABmag)
-        secondary = (catalog[self.secondary]*u.nJy).to_value(u.ABmag)
+        if isinstance(catalog, DataFrame) or isinstance(catalog, Table):
+            primary = np.array(catalog[self.primary])
+            secondary = np.array(catalog[self.secondary])
```

Comment:
I think this needs a comment saying why we need to convert to numpy here. I have a guess, but I'm not positive, so it should be spelled out.

Reply:
I'll add a comment.  It's because the DataFrame returns some sort of sequence that doesn't play nice with numpy operators unless you explicitly cast it.  I don't know why, it's a Pandas-is-Special thing.